### ISSUE TITLE
zc156: Downloads-handling issues

### ISSUE THREAD

**Issue by lat9** [zencart/zencart#1727]
I came across the following download-related issues debugging a related issue for _**One-Page Checkout**_:

## Issue 1: ##

```
PHP Warning:  A non-numeric value encountered in C:\xampp\htdocs\zc156\includes\classes\observers\auto.downloads_via_redirect.php on line 119
```

The affected line is
```php
      if ($exptime + SYMLINK_GARBAGE_COLLECTION_THRESHOLD > time()) continue;
```
I'm pretty sure that the issue is that the constant is not defined, but I'm not sure what value to use as the default.  If someone would give me some guidance here, I'll get a PR put together to address both issues.

## Issue 2: ##
```
PHP Fatal error:  1064:You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '1
                                     WHERE orders_id = 20
                  ' at line 3 :: UPDATE orders_products_download
                                     SET download_maxdays = 7,
                                         download_count  1
                                     WHERE orders_id = 20
                                     AND orders_products_download_id = 3 ==> (as called by) C:\xampp\htdocs\zc156\pincH-ael-prEss\orders.php on line 121 <== in C:\xampp\htdocs\zc156\includes\classes\db\mysql\query_factory.php on line 171
```
The affected line reads
```php
          $update_downloads_query = "UPDATE " . TABLE_ORDERS_PRODUCTS_DOWNLOAD . "
                                     SET download_maxdays = " . (int)$zc_max_days . ",
                                         download_count  " . (int)$chk_products_download_time->fields['products_attributes_maxcount'] . "
                                     WHERE orders_id = " . (int)$_GET['oID'] . "
                                     AND orders_products_download_id = " . (int)$_GET['download_reset_on'];
```
and should be updated to include a `=` after the _download_count_ field's name:
```php
          $update_downloads_query = "UPDATE " . TABLE_ORDERS_PRODUCTS_DOWNLOAD . "
                                     SET download_maxdays = " . (int)$zc_max_days . ",
                                         download_count = " . (int)$chk_products_download_time->fields['products_attributes_maxcount'] . "
                                     WHERE orders_id = " . (int)$_GET['oID'] . "
                                     AND orders_products_download_id = " . (int)$_GET['download_reset_on'];
```

**Comment by drbyte** [comment]
1: I think that constant on line 118 should be replaced with `$this->gc_cleanup_time` as defined on line 51. And then `$this->gc_cleanup_time` needs to be initialized as well, probably to `0`.

2: Agreed.

**Comment by drbyte** [comment]
Closed via #1735
